### PR TITLE
Delete button for admin

### PR DESCRIPTION
**Pull request by loneil**
# Description
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-1275

Add a Delete Submission button on the Admin Submission page.
Need to have "Editor" role (same as API endpoint) for this button to be enabled.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-vmvfjv-tools.pathfinder.gov.bc.ca/dashboard?id=common-forms-toolkit-pr-134

File: app/frontend/src/components/common/admin/inspection/DeleteButton.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      this.showDialog = true;
+    },
+    async deleteSubmission() {
+      try {
+        this.deleteError = '';
+        await commonFormService.removeSubmission(
+          this.formName,
+          this.submissionId
+        );
+        this.$router.push({ path: `/${this.formName}/admin` });
```
Comment (jujaga) [inline app/frontend/src/components/common/admin/inspection/DeleteButton.vue:70]: There should be a `this.showDialog = false` right before the router push. Probably "instantaneous", but we should probably make sure the dialog closes first before redirecting.

Comment (loneil) [reply app/frontend/src/components/common/admin/inspection/DeleteButton.vue:70]: 👍

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-vmvfjv-tools.pathfinder.gov.bc.ca/dashboard?id=common-forms-toolkit-pr-134

**Pull request merged**